core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -124,18 +127,30 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
                                       networkClient: NetworkClient,
                                       txnStateManager: TransactionStateManager,
                                       txnMarkerPurgatory: DelayedOperationPurgatory[DelayedTxnMarker],
-                                      time: Time) extends Logging {
+                                      time: Time,
+                                      metrics:Metrics) extends Logging {
+
+  private def addQueueLengthMetric(sensor: Sensor, namePrefix: String): Unit = {
+    sensor.add(metrics.metricName(s"$namePrefix-queue-length", "transaction-marker-channel-metrics"), new Total())
```

Comment:
Yes, I know the difference. I am stating that we use Yammer metrics for broker classes by default. For example, I just added a number of them for the Controller. I was trying to understand the reasoning for doing something different here.